src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2693 @@
+    type Output' Proto.NodeInfo'BakerConsensusInfo = BakerConsensusInfo
+    fromProto bcInfo = do
+        bakerId <- fromIntegral <$> bcInfo ^? ProtoFields.bakerId . ProtoFields.value
+        status <- do
+            st <- bcInfo ^. ProtoFields.maybe'status
+            case st of
+                ProtoFields.NodeInfo'BakerConsensusInfo'ActiveBakerCommitteeInfo' _ ->
+                    return ActiveBakerCommitteeInfo
+                ProtoFields.NodeInfo'BakerConsensusInfo'ActiveFinalizerCommitteeInfo' _ ->
+                    return ActiveFinalizerCommitteeInfo
+                ProtoFields.NodeInfo'BakerConsensusInfo'PassiveCommitteeInfo pcInfo ->
+                    PassiveBaker <$> fromProto pcInfo
+        return BakerConsensusInfo{..}
+
+data BakerConsensusInfo = BakerConsensusInfo
+    { bakerId :: !Integer
+    , status :: !BakerConsensusInfoStatus
+    }
+
+instance FromProto Proto.NodeInfo where
+    type Output' Proto.NodeInfo = NodeInfo
+    fromProto nInfo = do
+        nDetails <- nInfo ^. ProtoFields.maybe'details
+        let details = case nDetails of
+                ProtoFields.NodeInfo'Bootstrapper _ -> Nothing
+                ProtoFields.NodeInfo'Node' node -> do
+                    n <- node ^. ProtoFields.maybe'consensusStatus
+                    case n of
+                        ProtoFields.NodeInfo'Node'NotRunning _ ->
+                            return NodeNotRunning
+                        ProtoFields.NodeInfo'Node'Passive _ ->
+                            return NodePassive
+                        ProtoFields.NodeInfo'Node'Active cInfo ->
+                            NodeActive <$> fromProto cInfo
+        peerVersion <- nInfo ^? ProtoFields.peerVersion
+        localTime <- fmap fst . fromProto =<< nInfo ^? ProtoFields.localTime
+        peerUptime <- fromProto =<< nInfo ^? ProtoFields.peerUptime
+        networkInfo <- fromProto =<< nInfo ^? ProtoFields.networkInfo
+        return NodeInfo{..}
+
+data NodeDetails = NodeNotRunning | NodePassive | NodeActive !BakerConsensusInfo
+
+instance FromProto Proto.NodeInfo'NetworkInfo where
+    type Output' Proto.NodeInfo'NetworkInfo = NetworkInfo
+    fromProto nInfo = do
+        nodeId <- nInfo ^? ProtoFields.nodeId . ProtoFields.value
+        peerTotalSent <- fromIntegral <$> nInfo ^? ProtoFields.peerTotalSent
+        peerTotalReceived <- fromIntegral <$> nInfo ^? ProtoFields.peerTotalReceived
+        avgBpsIn <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsIn
+        avgBpsOut <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsOut
+        return NetworkInfo{..}
+
+data NetworkInfo = NetworkInfo
+    { nodeId :: !Text
+    , peerTotalSent :: !Integer
+    , peerTotalReceived :: !Integer
+    , avgBpsIn :: !Integer
+    , avgBpsOut :: !Integer
+    }
+
+data NodeInfo = NodeInfo
+    { peerVersion :: !Text
+    , localTime :: !Timestamp
+    , peerUptime :: !Duration
+    , networkInfo :: !NetworkInfo
+    , details :: !(Maybe NodeDetails)
```

Comment:
This is an odd choice here. Why not be explicit about being a bootstrapper or not?

Reply:
Do you mean something like:

```
data NodeInfo =
    NodeNotRunning { ... }
  | NodePassive { ... }
  | NodeActive { bakerConsensusInfo :: !BakerConsensusInfo, ... }
```

I suppose `NodePassive` means it is a bootstrapper (rather than a baker) then?